vertical-pod-autoscaler/pkg/recommender/model/cluster.go

```diff
@@ -143,9 +143,28 @@ func (cluster *ClusterState) AddOrUpdatePod(podID PodID, newLabels labels.Set, p
 
 		cluster.addPodToItsVpa(pod)
 	}
+	// Tally the number of containers for later when we're averaging the recommendations
+	cluster.setVPAContainersPerPod(pod)
 	pod.Phase = phase
 }
 
+// setVPAContainersPerPod sets the number of containers per pod seen for pods connected to this VPA
+// so that later when we're splitting the minimum recommendations over containers,  we're splitting them over
+// the correct number and not just the number of aggregates that have *ever* been present. (We don't want minimum resources
+// to erroneously shrink, either)
+func (cluster *ClusterState) setVPAContainersPerPod(pod *PodState) {
+	for _, vpa := range cluster.Vpas {
```

Comment:
I'm wondering if there is already a place where this logic could go so we don't have to loop over all VPAs for every pod again here.
In large clusters with a VPA to Pod ratio that's closer to 1 this could be a little wasteful.

Reply:
Hmm, yeah, I struggled with finding a less expensive way without making too much of a mess. Unless I'm missing something (and I might be) we don't seem to have a VPA <--> Pod map -- probably because we didn't need one until now? At the very least I think I should gate this to only run if the number of containers in the pod is > 1. 

Like, I think our options are:

1. update the VPA as the pods roll through (which requires me to find the VPA for each pod like I did here) or
2. count the containers as we load the VPAs (but we load the VPAs _before_ we load the pods, so we'd have to go through the pods _again_, so that doesn't help us)
3. have the VPA actually track the pods it's managing, something like this: https://github.com/jkyros/autoscaler/commit/6ddc2085905fcb2dafb5397822b9f893d436af67 (could also just be an array of `PodID`s and we could look up the state so we could save the memory cost of the PodState pointer, but you know what I mean) 

I put it where I did (`option 1`)  because at least `LoadPods()` was already looping through all the pods so we could freeload off the "outer" pod loop and I figured we didn't want to spend the memory on `option 3`. If we'd entertain `option 3` and are okay with the memory usage, I can totally do that?